lib/client.ml

```diff
@@ -1,10 +1,24 @@
 open Lwt_unix
 
+module Message = struct
+  type label =
+    | Request
+    | Response
+
+  type t = {
+    label : label;
+    sender : Peer.t;
+    payload : bytes;
+  }
+end
+
 type 'a t = {
   address : string;
   port : int;
   socket : file_descr;
   state : 'a ref;
+  request_inbox : Message.t Tqueue.t;
```

Comment:
What about `inbox` and `outbox`?

Reply:
`outbox` typically refers to a container of messages that are waiting to be sent :smile: neither of these really match that definition. However, I am not necessarily pleased with the current design of having one inbox field for every type of message.

Reply:
You are right, `outbox` does not fit at all. Do you see any other kind of messages? I mean a peer can either send `request` or receive `response` so it looks like we will be stuck with these two for the rest of the life, am I right?